pkg/volume/iscsi/iscsi.go

```diff
@@ -551,6 +551,17 @@ func createISCSIDisk(spec *volume.Spec, podUID types.UID, plugin *iscsiPlugin, m
 		return nil, err
 	}
 
+	if initiatorName != "" && len(bkportal) > 0 {
+		// Create a new iface and copy parameters from the pre-configured iface
+		newIface := bkportal[0] + ":" + spec.Name()
+		exec := plugin.host.GetExec(plugin.GetPluginName())
+		if err := cloneIface(exec, initiatorName, iface, newIface); err != nil {
```

Comment:
Is it possible to call `cloneIface` in `AttachDisk` where it originally was and only leave the old/new interface name calculation here? `createISCSIDisk` is called by `volumeSpecToMounter` and I think that does not need to clone the interface at all.

Reply:
I thought he was moving that to make the ref counter work, but maybe I am wrong about that.

Reply:
That's possible, but if we move the `cloneIface` back to `AttachDisk`, we'd have to check if the initiator is set in two different places (both here and in `AttachDisk`). It's not a big deal, but it seemed better to keep this logic together.

I don't have strong feelings about it, so I can move it back if you prefer.

Reply:
> That's possible, but if we move the cloneIface back to AttachDisk, we'd have to check if the initiator is set in two different places (both here and in AttachDisk)

I think it's OK.

I'm afraid that cloning interfaces in functions like `reconstructVolume`, `GenerateExpandInUseVolumeFunc` and `GetGlobalMapPath` won't do any good. It should be really used only when kubelet attaches & mounts a volume.

Reply:
I see; I had to add a new field to `iscsiDisk` to store the "old" iface (we need it to copy the parameters to the new one during attach).

The code may have got a bit tangled, but it was the way I found to clone ifaces only during attach.